### PR TITLE
Fixes Hit Escape moves interaction with hold effects and switch in ab…

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes Hit Escape move interaction with hold effects and switch in abilities

Fixes timing of hold effects with switching moves and Intimidate + Download. Both of them miss their activation timing. This was wrong in the expansion, including a couple of tests. All tested on showdown. For the review it would be good if some of it was tested in vanilla gen7+.

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -6954,13 +6934,16 @@ static void Cmd_switchineffects(void)
 
         // Don't activate switch-in abilities if the opposing field is empty.
         // This could happen when a mon uses explosion and causes everyone to faint.
-        if (ShouldPostponeSwitchInAbilities(battler) || gBattleStruct->switchInAbilityPostponed)
+        if ((battlerAbility == ABILITY_INTIMIDATE || battlerAbility == ABILITY_DOWNLOAD)
```
Comment (Bassoonian) [inline src/battle_script_commands.c:6937]: We need to make sure that when this PR makes it to upcoming, we apply the Intimidate changes to Supersweet Syrup too

File: src/battle_util.c [hunk 2]
```diff
@@ -4229,6 +4217,7 @@ u32 AbilityBattleEffects(u32 caseID, u32 battler, u32 ability, u32 special, u32
     u32 i, j;
     struct Pokemon *mon;
 
+
```
Comment (Bassoonian) [inline src/battle_util.c:4220]: random new empty line

File: test/battle/ability/intimidate.c [hunk 3]
```diff
@@ -159,3 +161,26 @@ DOUBLE_BATTLE_TEST("Intimidate activates on an empty slot")
         MESSAGE("Hitmontop's Intimidate cuts Foe Azurill's attack!");
     }
 }
+
+DOUBLE_BATTLE_TEST("Intimidate activates immediatly after the mon was switched in as long as one opposing mon is alive")
```
Comment (Bassoonian) [inline test/battle/ability/intimidate.c:165]:
```suggestion
DOUBLE_BATTLE_TEST("Intimidate activates immediately after the mon was switched in as long as one opposing mon is alive")
```

File: test/battle/move_effect/hit_escape.c [hunk 4]
```diff
@@ -112,3 +112,69 @@ SINGLE_BATTLE_TEST("U-turn switches the user out after Ice Face activates")
         MESSAGE("Go! Wynaut!");
     }
 }
+
+SINGLE_BATTLE_TEST("Held items are consumed immediately after a mon switched in by a bu U-turn and Intimidate activates after it: player side")
```
Comment (Bassoonian) [inline test/battle/move_effect/hit_escape.c:116]:
```suggestion
SINGLE_BATTLE_TEST("Held items are consumed immediately after a mon switched in by U-turn and Intimidate activates after it: player side")
```

File: src/battle_script_commands.c (hunk 1 above)
Comment (AlexOn1ine) [reply src/battle_script_commands.c:6937]: Already tried to include it until I realized this goes to master. Will make sure to take care of it.

File: src/battle_util.c (hunk 2 above)
Comment (AlexOn1ine) [reply src/battle_util.c]: fixed

File: test/battle/ability/intimidate.c (hunk 3 above)
Comment (AlexOn1ine) [reply test/battle/ability/intimidate.c]: fixed

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**